src/main/java/net/bootsfaces/listeners/AddResourcesListener.java

```diff
@@ -73,88 +73,138 @@
 
 	/**
-	 * Components can request themed CSS resources by registering them in the
-	 * ViewMap, using the THEME_RESOURCE_KEY.
+	 * Components can request themed CSS resources by registering them in the ViewMap, using the THEME_RESOURCE_KEY.
 	 */
 	private static final String THEME_RESOURCE_KEY = "net.bootsfaces.listeners.AddResourcesListener.ThemedResourceFiles";
 
 	/**
-	 * Components can request extended CSS resources by registering them in the
-	 * ViewMap, using the EXT_RESOURCE_KEY.
+	 * Components can request extended CSS resources by registering them in the ViewMap, using the EXT_RESOURCE_KEY.
 	 */
 	private static final String EXT_RESOURCE_KEY = "net.bootsfaces.listeners.AddResourcesListener.ExtResourceFiles";
 
 	private static final String SCRIPT_RENDERER = "javax.faces.resource.Script",
-			CSS_RENDERER = "javax.faces.resource.Stylesheet";
+		CSS_RENDERER = "javax.faces.resource.Stylesheet";
 
 	static {
 		LOGGER.info("This application is running on BootsFaces " + C.BSFVERSION + "-" + C.BSFRELEASE_STATUS);
 	}
 
 	/**
 	 * Returns true if the source is an instance of {@link UIViewRoot}.
+	 *
+	 * @param source
+	 * @return
 	 */
 	@Override
 	public boolean isListenerForSource(Object source) {
 		return (source instanceof UIViewRoot);
 	}
 
 	/**
-	 * Trigger adding the resources if and only if the event has been fired by
-	 * UIViewRoot.
+	 * Trigger adding the resources if and only if the event has been fired by UIViewRoot.
+	 *
+	 * @param event
 	 */
+	@Override
 	public void processEvent(SystemEvent event) throws AbortProcessingException {
 		FacesContext context = FacesContext.getCurrentInstance();
 		UIViewRoot root = context.getViewRoot();
 
 		// render the resources only if there is at least one bsf component
-		if (ensureExistBootsfacesComponent(root, context)) {
+		if (ensureExistBootsfacesComponent(root)) {
 			addCSS(root, context);
 			addJavascript(root, context);
 			addMetaTags(root, context);
 		}
 	}
 
 	/**
-	 * Check if there is almost one bootsfaces component in page. If yes, load the
-	 * correct items.
 	 *
-	 * To ensure that, it checks first the viewMap to find specific bootsfaces key.
-	 * If it found nothing, it check for components that has as a resource lib, the
-	 * "bsf" lib.
+	 * @param root the UIViewRoot
+	 * @return
+	 */
+	private static Map<String, Object> getMapForView(UIViewRoot rootView) {
+		// if the view is transient, we use only the attribute-store
+		if (rootView.isTransient()) {
+			LOGGER.log(Level.FINEST, "use attribute-store");
+			return rootView.getAttributes();
+		} else {
+			// if a session exists, we can use the viewMap-store
+			// check if a session exists
+			boolean sessionExists = FacesContext.getCurrentInstance().getExternalContext().getSession(false) != null;
+			if (sessionExists) {
+				LOGGER.log(Level.FINEST, "use viewMap-store");
+				return rootView.getViewMap();
+			} else {
+				// as far as I know, a session is started if there is a UIForm
```

Comment:
I don't think this is true (at least in myfaces 2.3) http://www.icesoft.org/JForum/posts/list/22733.page